nixd/include/nixd/Protocol/AttrSet.h

```diff
@@ -0,0 +1,52 @@
+/// \file
+/// \brief Types used in nixpkgs provider.
+
+#pragma once
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <llvm/Support/JSON.h>
+
+namespace nixd {
+
+using EvalExprParams = std::string;
+using EvalExprResponse = std::optional<std::string>;
+
+using AttrPathInfoParams = std::string;
```

Comment:
I would like to use vector for this argument. It is more general and less error prone.